.github/workflows/ci.yml

```diff
@@ -1,5 +1,9 @@
 name: CI
-on: [push]
+on:
+  push:
+    branches: [ main ]
+  pull_request:

```

Comment:
since we initially missed this one, I'll have to make a PR from a branch on this repo instead of from my fork I believe.

I'll drop this PR and switch to that so we can ensure the action runs correctly, and once this is merged, forks should get CI properly too.